storyruntime/utils/HttpUtils.py

```diff
@@ -53,4 +54,18 @@ def add_params_to_url(url, params: dict):
         if len(params) == 0:
             return url
 
-        return f'{url}?{urlencode(params)}'
+        parts = list()
+
+        for key, value in params.items():
+            key = urllib.parse.quote(key)
+            if type(value) is list:
+                for lv in value:
+                    parts.append(
+                        f'{key}={urllib.parse.quote(str(lv))}'
+                    )
+            else:
+                parts.append(
+                    f'{key}={urllib.parse.quote(str(value))}'
+                )
+
+        return f'{url}?{("&".join(parts))}'
```

Comment:
Let's use urlencode instead here please:

https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode

> Convert a mapping object or a sequence of two-element tuples, which may contain str or bytes objects, to a percent-encoded ASCII text string

> The resulting string is a series of key=value pairs separated by '&' characters, where both key and value are quoted using the quote_via function. By default, quote_plus() is used to quote the values, which means spaces are quoted as a '+' character and ‘/’ characters are encoded as %2F, which follows the standard for GET request

```suggestion
        return url + urllib parse.urlencode(parts)
```